src/server/db/localstorage.ts

```diff
@@ -0,0 +1,61 @@
+import { InMemory } from './inmemory';
+
+class WithLocalStorageMap<Tkey, Tvalue> extends Map {
+  key: string;
+  constructor(key: string) {
+    super();
+    const inCache = initFromCache(key);
+    inCache.forEach((pair: any[]) => {
+      this.set(pair[0], pair[1]);
+    });
+    this.key = getStorageKey(key);
+  }
+  sync(): void {
+    localStorage.setItem(this.key, JSON.stringify(Array.from(this.entries())));
```

Comment:
`this.key` is `undefined `on the first call before even the game inits
should add something like
```
if (!this.key) {
  return;
}
```

Reply:
You are right. Is fixed.